docs/appendices/release-notes/unreleased.rst

```diff
@@ -56,6 +56,9 @@ None
 Changes
 =======
 
+- Improved the handling of constant join criteria to make sure it can be
+  pushed to the collect phase when possible. 
+
```

Comment:
I think this could be rephrased to highlight the user facing impact more and focus less on the technical aspect.

"to make sure it can be pushed to the collect phase when possible. " states what happens internally but it doesn't communicate the impact the user will observe.